source/grid/manifold_lib.cc

```diff
@@ -1376,13 +1396,16 @@ FunctionManifold<dim, spacedim, chartdim>::clone() const
   // push forward and the pull back charts, or by providing two Function
   // objects. In the first case, the push_forward and pull_back functions are
   // created internally in FunctionManifold, and destroyed when this object is
-  // deleted. We need to make sure that our cloned object is constructed in the
+  // deleted. In the second case, the function objects are destroyed if they
+  // are passed as pointers upon construction.
+  // We need to make sure that our cloned object is constructed in the
   // same way this class was constructed, and that its internal Function
   // pointers point either to the same Function objects used to construct this
-  // function (owns_pointers == false) or that the newly generated manifold
-  // creates internally the push_forward and pull_back functions using the same
-  // expressions that were used to construct this class (own_pointers == true).
-  if (owns_pointers == true)
+  // function or that the newly generated manifold creates internally the
+  // push_forward and pull_back functions using the same expressions that were
+  // used to construct this class.
+  if (const FunctionParser<chartdim> *f =
+        dynamic_cast<const FunctionParser<chartdim> *>(&*push_forward_function))
```

Comment:
What if one of the constructors taking `Function` objects (instead of `std::string` objects) is a `FunctionParser` object?
In this case, we are trying to create an object with an empty expression string. Hence, I would rather check for `push_forward_expression` and `pull_back_expression` being empty.